pkg/cli/serve/config.go

```diff
@@ -0,0 +1,40 @@
+package serve
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+// ReadFlagsFromFile checks for '@' prefix, if found try to read value from file.
+func ReadFlagsFromFile(cmd *cobra.Command, names ...string) error {
+	for _, name := range names {
+		value, err := cmd.Flags().GetString(name)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "unable to fetch value for key %s\n", name)
+			os.Exit(1)
+		}
+		if strings.HasPrefix(value, "@") {
```

Comment:
why we have to have a `@` in each of the values?

Reply:
This was something I requested as part of the OSD-9709 acceptance criteria.

If the argument is preceded by a "@" the agent infers the value supplied via the CLI is a filepath which contains the config value, and reads the value from there. 

If the argumnt is not, it infers the value supplied on the CLI. 

The former behaviour allows for the ability to mount a ConfigMap through to the container and read files in the way that configmap keys/values get mounted. The latter behaviour is easier for local testing.

Cluster Service works in a very similar way.